.circleci/config.yml

```diff
@@ -143,36 +170,49 @@ jobs:
           name: 'Validator Tests'
           command: node build-system/pr-check/validator-tests.js
       - fail_fast
+      - teardown_vm
   'Visual Diff Tests':
     executor:
       name: amphtml-large-executor
     steps:
+      # NOTE: deliberately not running the `maybe_gracefully_halt` step because
+      # visual diff tests need to at least run with --empty when skipped.
+      # Instead we must create an empty workspace to prevent `teardown_vm` from
+      # crashing.
```

Comment:
I recognize that this special case exists because Percy is a required check. However, this looks like a detail of implementation of visual testing, and I think it should live in nodejs code and not in the main CircleCI config file.

Alternative suggestion: Move the call to `timedExecOrDie('amp visual-diff --empty');` from `pr-check/visual-diff-tests.js` to this section of `pr-check/nomodule-build.js` (say, after line 70). 

https://github.com/ampproject/amphtml/blob/20813ccd2bcdc7c80defb601081c7509ef2455c0/build-system/pr-check/nomodule-build.js#L69-L75

With this change, we can gracefully halt the `Visual Diff Tests` job without having to pay the additional task of VM setup just to create a blank Percy build, and without this duplicate `Create Workspace` step.

Reply:
Makes sense! I'll do that

Reply:
Done

Reply:
@danielrozenberg We forgot one thing. Skipping a Percy build also needs the token. See [this broken build](https://app.circleci.com/pipelines/github/ampproject/amphtml/6229/workflows/0d129b3f-f949-42df-bb67-e2ec2dbdbc69/jobs/88512/parallel-runs/0/steps/0-111).

We'll need to insert this code:

https://github.com/ampproject/amphtml/blob/d71ab5009757675e476b7f3ea26bfd97ef1d8d3d/build-system/pr-check/visual-diff-tests.js#L40

PR incoming.